File: dSoak-Common-CSharp/SharedObjectTesting/GameInfoTester.cs
using System;
using Microsoft.VisualStudio.TestTools.UnitTesting;

using SharedObjects;


namespace SharedObjectTesting
{
    [TestClass]
    public class GameInfoTester
    {
        [TestMethod]
        public void GameInfo_TestEverything()
        {
            GameInfo g1 = new GameInfo();
            Assert.AreEqual(0, g1.FightManagerId);
            Assert.AreEqual(null, g1.FightManagerEP);
            Assert.AreEqual(0, g1.GameId);
            Assert.AreEqual(0, g1.MaxPlayers);
            Assert.AreEqual(GameInfo.StatusCode.NotInitialized, g1.Status);

            PublicEndPoint ep1 = new PublicEndPoint() { Host = "buzz.serv.usu.edu", Port = 20011 };
            GameInfo g2 = new GameInfo() { FightManagerId=2, FightManagerEP = ep1, GameId = 10, Label="Test Game", MaxPlayers = 5, MaxThieves=2, Status = GameInfo.StatusCode.NotInitialized };
            Assert.AreEqual(2, g2.FightManagerId);
            Assert.AreEqual(ep1, g2.FightManagerEP);
            Assert.AreEqual(10, g2.GameId);
            Assert.AreEqual("Test Game", g2.Label);
            Assert.AreEqual(5, g2.MaxPlayers);
            Assert.AreEqual(2, g2.MaxThieves);
            Assert.AreEqual(GameInfo.StatusCode.NotInitialized, g2.Status);

            GameInfo g3 = g2.Copy;
            Assert.IsNotNull(g3);
            Assert.AreNotSame(g2, g3);
            Assert.AreEqual(2, g3.FightManagerId);
            Assert.AreEqual(ep1, g3.FightManagerEP);
            Assert.AreEqual(10, g3.GameId);
            Assert.AreEqual("Test Game", g3.Label);
            Assert.AreEqual(5, g3.MaxPlayers);
            Assert.AreEqual(2, g3.MaxThieves);
            Assert.AreEqual(GameInfo.StatusCode.NotInitialized, g3.Status);
        }

    }
}

File: dSoak-Common-Java/src/SharedObjectTester/GameInfoTester.java
package SharedObjectTester;

import static org.junit.Assert.*;

import java.net.UnknownHostException;

import org.junit.Test;

import SharedObject.GameInfo;
import SharedObject.PublicEndPoint;

public class GameInfoTester  
{
	@Test
	public void test_EveryThing() throws UnknownHostException 
	{
		GameInfo g1 = new GameInfo();
		
		assertEquals(null, g1.FlightManagerEP);
		assertEquals(0, g1.GameId);
		assertEquals(0, g1.MaxPlayers);
		assertNull(g1.Status);
		
		PublicEndPoint ep1 = new PublicEndPoint("buzz.serv.usu.edu:20011");
		
		GameInfo g2 = new GameInfo();
		g2.FlightManagerEP = ep1;
		g2.FightManagerId = 2;
		g2.GameId = 10;
		g2.MaxPlayers = 5;
		g2.Status = GameInfo.StatusCode.AVAILABLE;
		g2.MaxThieves = 2;
		
		assertTrue(ep1.getHost().equals(g2.FlightManagerEP.getHost()));
		assertEquals(2, g2.FightManagerId);
		assertEquals(10, g2.GameId);
		assertEquals(5,g2.MaxPlayers);
		assertEquals(GameInfo.StatusCode.AVAILABLE, g2.Status);
		assertEquals(2, g2.MaxThieves);
	}
}
